chpl/chpl-service/src/main/java/gov/healthit/chpl/app/GenerateDailyChartData.java

```diff
@@ -0,0 +1,200 @@
+		} else {
+			props = new Properties();
+			props.load(in);
+			in.close();
+		}
+
+		//set up data source context
+		LocalContext ctx = LocalContextFactory.createLocalContext(props.getProperty("dbDriverClass"));
+		ctx.addDataSource(props.getProperty("dataSourceName"),props.getProperty("dataSourceConnection"), 
+				props.getProperty("dataSourceUsername"), props.getProperty("dataSourcePassword"));
+
+		//init spring classes
+		AbstractApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+		GenerateDailyChartData app = new GenerateDailyChartData();
+		app.setListingStatisticsDao((ListingStatisticsDAO)context.getBean("listingStatisticsDAO"));
+		app.setChartDataDao((ChartDataDAO)context.getBean("chartDataDAO"));
+		app.setCertificationStatusEventDao((CertificationStatusEventDAO)context.getBean("certificationStatusEventDAO"));
+		app.setCertificationEditionDao((CertificationEditionDAO)context.getBean("certificationEditionDAO"));
+		List<List<String>> doubleList = new ArrayList<List<String>>();
+		LocalDate begin = LocalDate.of(2016, Month.APRIL, 1);
+		LocalDate end = LocalDate.now();
+		LocalDate current = begin;
+		while(current.isBefore(end) || current.isEqual(end)){
+			Date date1 = java.sql.Date.valueOf(begin);
+			Date date2 = java.sql.Date.valueOf(current);
+			DateRange dateRange = new DateRange(date1,date2);
+			Long uniqueProductsOverTime = app.getListingStatisticsDao().getTotalCertifiedProducts(dateRange);
+			Long uniqueProductsActive = app.getListingStatisticsDao().getTotalCPsActiveListings(dateRange);
+			Long uniqueProductsActive2014 = app.getListingStatisticsDao().getTotalActive2014Listings(dateRange);
+			Long uniqueProductsActive2015 = app.getListingStatisticsDao().getTotalActive2015Listings(dateRange);
+			List<String> array = new ArrayList<String>();
+			array.add(current.toString());
+			array.add(uniqueProductsOverTime.toString());
+			array.add(uniqueProductsActive.toString());
+			array.add(uniqueProductsActive2014.toString());
+			array.add(uniqueProductsActive2015.toString());
+			current = current.plusDays(1);
+			doubleList.add(array);
+		}
+		System.out.println(doubleList);
+		Date date1 = java.sql.Date.valueOf(begin);
+		List<Long> certifiedIdsWithChangedHistory = app.getListingStatisticsDao().getProductsWithChangedCertificationStatusPostApril2016(date1);
+		System.out.println(certifiedIdsWithChangedHistory.size());
+		for(Long id : certifiedIdsWithChangedHistory){
+			List<CertificationStatusEventDTO> events = app.getCertificationStatusEventDao().findByCertifiedProductId(id);
+			for(int i=0;i<events.size()-1;i++){
+				if(events.get(i).getEventDate().after(date1)){
+					Date beginDate = events.get(i).getCreationDate();
+					Date endDate = events.get(i+1).getCreationDate();
+					// if going from status 1 to something else decrement all dates in between begin and end
+					if(events.get(i).getStatus().getId().equals(1L)){
```

Comment:
It might just be that I'm not as familiar with this code but I'm not clear on exactly what this is doing (I can see it's loading daily statistics but I'm more lost with the getting and setting of the list variable below). Could you add a comment block to explain a little bit? Or maybe use some helpfully-named constants here instead of 0, 1, 3, 4?